File: client/src/main/java/com/jraska/github/client/ui/ErrorHandler.java
package com.jraska.github.client.ui;

import android.support.v7.widget.RecyclerView;

import com.airbnb.epoxy.EpoxyModel;
import com.airbnb.epoxy.SimpleEpoxyAdapter;

import java.net.UnknownHostException;
import java.util.Collections;
import java.util.List;

public final class ErrorHandler {
  @SuppressWarnings("unchecked")
  List<EpoxyModel<?>> modelsForError(Throwable error) {
    if (error instanceof UnknownHostException) {
      return Collections.singletonList(ErrorModel.networkError());
    }

    return Collections.singletonList(ErrorModel.genericError());
  }

  static void displayError(Throwable throwable, RecyclerView toView) {
    ErrorHandler errorHandler = new ErrorHandler();
    List<EpoxyModel<?>> models = errorHandler.modelsForError(throwable);

    SimpleEpoxyAdapter epoxyAdapter = new SimpleEpoxyAdapter();
    epoxyAdapter.addModels(models);
    toView.setAdapter(epoxyAdapter);
  }
}

File: client/src/main/java/com/jraska/github/client/ui/ErrorHandler.kt
package com.jraska.github.client.ui

import androidx.recyclerview.widget.RecyclerView
import com.airbnb.epoxy.EpoxyModel
import com.airbnb.epoxy.SimpleEpoxyAdapter
import java.net.UnknownHostException

object ErrorHandler {
  fun modelsForError(error: Throwable): List<EpoxyModel<*>> {
    if (error is UnknownHostException) {
      return listOf(ErrorModel.networkError())
    }

    return listOf(ErrorModel.genericError())
  }

  fun displayError(throwable: Throwable, toView: RecyclerView) {
    val models = modelsForError(throwable)

    val epoxyAdapter = SimpleEpoxyAdapter()
    epoxyAdapter.addModels(models)
    toView.adapter = epoxyAdapter
  }
}
